### PR TITLE
Enhancement to percentiles endpoint for supporting multiple year range - WPS #95

### PR DESCRIPTION
**Pull request by Nkuruba**
As part of wps #95 Change added to wps-api to support requests with data ranges 20 and full.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,10 @@ If you have trouble getting pipenv to resolve python 3.8.1, you can also try:
 ```
 pipenv install --python ~/.pyenv/versions/3.6.10/bin/python3.6 --dev
 ```
+If you have trouble installing pyodbc, you can try:
```
Comment (Sybrand) [inline README.md]: This comment us under the mac os heading but specific to Ubuntu - please put it under an appropriate heading, or clarify that it relates directly to ubuntu.

Comment (Nkuruba) [reply README.md]: Unit tests fixed.
Updated readme.
Pushed updates to script.

File: main.py [hunk 2]
```diff
@@ -87,12 +87,20 @@ class WeatherStations(BaseModel):
     """
     # NOTE: percentile, start_year and end_year input is ignored, all responses overriden to match
     # pre-calculated values; 90th percentile, start year 2010, end year 2019
-    response = CalculatedResponse(percentile=90, year_range=YearRange(start=2010, end=2019))
+    default_year_range = YearRange(start=2010, end=2019)
```
Comment (Sybrand) [inline main.py]: We actually no longer need a default year range - we can now make it required - and use the actual year range supplied. This was just a workaround I introduced before we had year ranges.

Comment (Nkuruba) [reply main.py]: Removed default year range.

File: main.py [hunk 3]
```diff
@@ -87,12 +87,19 @@ class WeatherStations(BaseModel):
     """
     # NOTE: percentile, start_year and end_year input is ignored, all responses overriden to match
```
Comment (andrea-williams) [inline main.py]: this comment should be updated - we are now only overriding the percentile

File: scripts/calculate_percentile_offline.py [hunk 4]
```diff
@@ -133,7 +136,10 @@ def write_output_to_json():
                 daily_weather_data.loc[daily_weather_data['station_code']
                                        == index, 'station_name'].iloc[0]
         }
-        output_filename = "../data/" + str(index) + ".json"
+        output_folder = "../data/" + date_range[0:4] + "-" + date_range[5:9]
```
Comment (andrea-williams) [inline scripts/calculate_percentile_offline.py]:
```suggestion
        output_folder = "../data/" + date_range
```

Comment (Nkuruba) [reply scripts/calculate_percentile_offline.py]: Added the change.

File: scripts/calculate_percentile_offline.py [hunk 5]
```diff
@@ -37,13 +39,14 @@
...
 # the algorithm
 def main():
-    parse_weather_dates()
-    remove_data_outside_date_range()
-    remove_data_outside_fire_season()
-    sort_by_weather_station()
-    calculate_percentile_per_station()
-    list_years_per_station()
-    write_output_to_json()
+    for item in range(len(RANGES)):
```
Comment (Sybrand) [inline scripts/calculate_percentile_offline.py]: Should be able to do
```suggestion
    for item in RANGES:
```

File: scripts/calculate_percentile_offline.py [hunk 6]
```diff
@@ -37,13 +39,14 @@
...
+        parse_weather_dates()
+        remove_data_outside_date_range(RANGES[item])
```
Comment (Sybrand) [inline scripts/calculate_percentile_offline.py]: then here, do:
```suggestion
        remove_data_outside_date_range(item)
```

File: scripts/calculate_percentile_offline.py [hunk 7]
```diff
@@ -10,9 +11,10 @@
 FIRE_SEASON_END_DATE = 31
 
 # time range start and end years
+# start of 1970 fire season to end of 2019 fire season is 50 years
+# start of 2000 fire season to end of 2019 fire season is 20 years
 # start of 2010 fire season to end of 2019 fire season is 10 years
-START_YEAR = 2010
-END_YEAR = 2019
+RANGES = ["1970-2019", "2000-2019", "2010-2019"]
```
Comment (Sybrand) [inline scripts/calculate_percentile_offline.py]: I would have used
```suggestion
RANGES = [[1970, 2019], [2000, 2019], [2010-2019]]
```

**Pull request merged**